changes against list of canonical skills: qa-knowledge, qa-structure, data-collection (#128)
griddynamics/rosetta: instructions/r3/core/skills/qa-knowledge/SKILL.md

---
name: qa-knowledge
description: "To supply the QA-domain conventions: failure taxonomies, authoring & correction discipline, and artifact skeletons."
license: Apache-2.0

> disable-model-invocation: false

Reviewer: ```suggestion
disable-model-invocation: true
```

user-invocable: false

> tags: []

Reviewer: ```suggestion
```

baseSchema: docs/schemas/skill.md
---

<qa_knowledge>

<when_to_use_skill>

> Use when authoring, analyzing, or correcting backend-API or UI/E2E tests and needing QA conventions: failure taxonomies, assertion & coverage discipline, selector & page-object rules, and the artifact skeletons these tasks emit.

Reviewer: ```suggestion
Use when authoring, analyzing, or correcting backend-API or UI/E2E tests and needing QA conventions: failure taxonomies, assertion & coverage discipline, selector & page-object rules, and the artifact skeletons these tasks emit. TestRail/Jira/Confluence are used as canonical examples, adapt to current case.
```

</when_to_use_skill>

<core_concepts>

- Load only what the current task needs;

> - artifact skeletons are assets, ACQUIRE'd at point of use;

Reviewer: ```suggestion
- artifact skeletons are assets, ACQUIRE FROM KB at point of use;
```

- conventions are references -- see `<resources>`.

</core_concepts>

<resources>

> Router -- ACQUIRE the one your current step needs (point-of-use, never all at once):

Reviewer: ```suggestion
Router -- ACQUIRE FROM KB the one your current step needs (point-of-use, never all at once):
```

> | When you need to… | ACQUIRE |

Reviewer: ```suggestion
| When you need to… | ACQUIRE ... FROM KB |
```
